### PR TITLE
Add comments explaining the sync files and their repsonsibilities

### PR DESCRIPTION
**Pull request by david-martin**

### CODE REVIEW

**Comment by david-martin** [comment]
@wei-lee would you mind sanity checking these comments for accuracy?
Thanks

File: lib/sync-srv.js [hunk 1]
```diff
@@ -16,6 +16,31 @@ module.exports = function (cfg) {
 
 var syncInited = false;
 
+/**
+ * This is the main export that exposes the fh.sync API.
+ * An instance of the sync API is created when this function is called.
+ * See api.js for where the instance is wired up and exposed as fh.sync.
+ *
```
Comment (fh-qube) [inline lib/sync-srv.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

File: lib/sync-srv.js [hunk 2]
```diff
@@ -16,6 +16,31 @@ module.exports = function (cfg) {
...
+ * -- responds to the client with details of all processed updates, and their status
+ * - sync-DataSetModel.js
+ * -- Runs the sync loop (with a 500ms interval between the end of 1 loop and start of next)
+ * -- Calls out to the data handlers (in sync-datahander.js by default) if a particular DataSet is scheduled to be synced
+ * - sync-datahandler.js
+ * -- Default data handler implementation for the DataSet Backend. Uses fh.db to store the DataSets in mongodb.
+ * -- Not used if the developer overwrites all the data handlers
+ * - sync-util.js
+ * -- Has utility functions for generating a hash of an object, sorting a json object by key, logging (and a couple of others)
+ *
```
Comment (fh-qube) [inline lib/sync-srv.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by david-martin** [comment]
Closing this as sync horizontal scalability work will supersede the existing sync implementation https://issues.jboss.org/browse/FH-3123

**Pull request closed without merging**